src/signing.rs

```diff
@@ -1,10 +1,16 @@
 //! Signing capabilities and utilities.
 
-use crate::types::{Address, H256};
+#[cfg(feature = "signing")]
+use crate::types::Address;
+use crate::types::H256;
+#[cfg(feature = "signing")]
 use secp256k1::recovery::{RecoverableSignature, RecoveryId};
+#[cfg(feature = "signing")]
 use secp256k1::{Message, PublicKey, Secp256k1};
+#[cfg(feature = "signing")]
 use std::ops::Deref;
 
+#[cfg(feature = "signing")]
```

Comment:
And here as well. Perhaps it's better to simply disable the entire `signing.rs` file?